src/redux/books/booksSlice.js

```diff
@@ -1,39 +1,71 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { v4 as uuid } from 'uuid';
+import ApiBooks from '../booksApI';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  loading: false,
+  books: [],
+  error: '',
+  state: 'idle',
 };
 
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await ApiBooks.get('/books');
+  return response.data;
+});
+
+export const addBook = createAsyncThunk(
+  'books/createBook',
+  async (bookData) => {
+    const response = await ApiBooks.post('/books', bookData);
+    if (response.data === 'Created') {
+      // Book create successfuly
+    }
+  },
+);
+
+export const deleteBook = createAsyncThunk('books/deleteBook', async (id) => {
+  await ApiBooks.delete(`/books/${id}`);
+  return id;
+});
+
 const booksSlice = createSlice({
   name: 'books',
   initialState,
-  reducers: {
-    addBooks: (state, action) => {
-      state.books.push(action.payload);
-    },
-    removeBook: (state, action) => {
-      // eslint-disable-next-line no-param-reassign
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchBooks.pending, (state) => {
+      state.loading = true;
+    });
+    builder.addCase(fetchBooks.fulfilled, (state, action) => {
+      state.loading = false;
+      state.books = Object.entries(action.payload).map(
+        ([itemId, itemData]) => ({
+          item_id: itemId,
+          ...itemData[0],
+        }),
+      );
+      state.error = '';
+    });
+    builder.addCase(fetchBooks.rejected, (state, action) => {
+      state.loading = false;
+      state.books = [];
+      state.error = action.error.message;
+    });
+    builder.addCase(addBook.fulfilled, (state, action) => {
+      state.status = 'succeeded';
+      const book = {
+        item_id: uuid(),
+        ...action.meta.arg,
+      };
+      state.books = [...state.books, book];
+      state.loading = true;
+    });
+    builder.addCase(deleteBook.fulfilled, (state, action) => {
+      state.status = 'succeeded';
       state.books = state.books.filter((book) => book.item_id !== action.payload);
-    },
+    });
```

Comment:
- [ ]  You did a great job to work on the project with external API but there some issues that need to be fixed for the app to work properly. 

- Creating a book is loading for forever, and I have to reload the page, so that I can see the created book, please fix it so that I don't need to reload to see the created book 
- Deleting a book is providing message in UI saying that `something is wrong`. So that I have to reload the page to see the deleted book.  fix these issues so that I don't need to reload the page to see the updated page. 
![image](https://github.com/danielochuba/bookstore/assets/43877689/cd9c5a68-e5f8-40b3-b428-bded250b92ac)